integration_tests/src/main/python/asserts.py

```diff
@@ -26,23 +26,26 @@
 import difflib
 import sys
 
-def _assert_equal(cpu, gpu, float_check, path):
+def _assert_equal(cpu, gpu, float_check, path, cpu_data_adapted_from = "CPU"):
+    """
+    :param cpu_data_adapted_from: indicate where the cpu result is adapted from, e.g.: pyarrow
+    """
     t = type(cpu)
     if (t is Row):
-        assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
+        assert len(cpu) == len(gpu), "{} and GPU row have different lengths at {} CPU: {} GPU: {}".format(cpu_data_adapted_from, path, len(cpu), len(gpu))
         if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
-            assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
+            assert cpu.__fields__ == gpu.__fields__, "{} and GPU row have different fields at {} CPU: {} GPU: {}".format(cpu_data_adapted_from, path, cpu.__fields__, gpu.__fields__)
```

Comment:
Here too. Do we want to replace `CPU:` with `{}:`? This also applies to like 44 and 48

Reply:
Done.